src/test/java/org/folio/rest/api/AbstractInstanceRecordsAPITest.java

```diff
@@ -0,0 +1,124 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import javax.ws.rs.core.Response;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.api.TestBase;
+import org.folio.rest.impl.AbstractInstanceRecordsAPI;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.AdditionalAnswers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+@RunWith(VertxUnitRunner.class)
+public class AbstractInstanceRecordsAPITest extends TestBase {
+  private static Map<String,String> okapiHeaders = Collections.singletonMap(
+      RestVerticle.OKAPI_HEADER_TENANT, StorageTestSuite.TENANT_ID);
+
+  class MyAbstractInstanceRecordsAPI extends AbstractInstanceRecordsAPI {
+    public void fetchRecordsByQuery(String sql, RoutingContext routingContext,
+        Supplier<Tuple> paramsSupplier, Handler<AsyncResult<Response>> asyncResultHandler) {
+
+      fetchRecordsByQuery(sql, paramsSupplier, routingContext, okapiHeaders,
+          asyncResultHandler, Vertx.vertx().getOrCreateContext(), null);
+    }
+  }
+
+  @Test
+  public void shouldRespondWith500StatusWhenErrorsOccursWhilstFetchingRecords(TestContext testContext) {
+    RoutingContext routingContext = mock(RoutingContext.class);
+    when(routingContext.response()).thenReturn(mock(HttpServerResponse.class));
+    new MyAbstractInstanceRecordsAPI().fetchRecordsByQuery("SELECT 1",
+        routingContext, null, testContext.asyncAssertSuccess(response -> {
+          assertThat(response.getStatus(), is(500));
+        }));
+  }
+
+  @Test
+  public void shouldCloseTcpWhenFailureAfterHttpHeadHasBeenWritten(TestContext testContext) {
+    HttpServerResponse httpServerResponse = mock(HttpServerResponse.class);
+    when(httpServerResponse.headWritten()).thenReturn(true);
+    RoutingContext routingContext = mock(RoutingContext.class);
+    when(routingContext.response()).thenReturn(httpServerResponse);
+    new MyAbstractInstanceRecordsAPI().fetchRecordsByQuery("SELECT 1",
+        routingContext, null, testContext.asyncAssertSuccess(response -> {
+          assertThat(response, is(nullValue()));
+          verify(httpServerResponse).close();
+        }));
+  }
+
+  private HttpServerResponse getHttpServerResponseMock() {
+    HttpServerResponse httpServerResponse = mock(HttpServerResponse.class);
+    doAnswer(AdditionalAnswers.answerVoid(
+        (Handler<AsyncResult<Void>> handler) -> handler.handle(Future.succeededFuture())))
+        .when(httpServerResponse).end(any(Handler.class));
+    return httpServerResponse;
+  }
```

Comment:
Personally, I would move this method to the bottom. In test classes I prefer that all of the tests are listed first, as I consider those to be the important parts of the class.